src/components/PostsList.tsx

```diff
@@ -40,7 +40,7 @@
                   'is-light': post.id !== selectedPostId,
                 })}
                 onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
+                  setSelectedPost(post.id === selectedPostId ? null : post);
```

Comment:
The logic here correctly toggles the selected post. If the current post is selected, it sets the selected post to null, effectively deselecting it. Otherwise, it sets the selected post to the current post. Ensure that the 'setSelectedPost' function is implemented to handle these changes appropriately.